src/core/dense_set.cc

```diff
@@ -255,6 +268,39 @@ void DenseSet::CloneBatch(unsigned len, CloneItem* items, DenseSet* other) const
   }
 }
 
+void DenseSet::ClearBatch(unsigned len, ClearItem* items) {
+  while (len) {
+    unsigned dest_id = 0;
+    // we walk "len" linked lists in parallel, and prefetch their next, obj pointers
+    // before actually processing them.
+    for (unsigned i = 0; i < len; ++i) {
+      auto& src = items[i];
+      if (src.obj) {
+        ObjDelete(src.obj, src.has_ttl);
+        src.obj = nullptr;
+      }
+
+      if (src.ptr.IsEmpty())
+        continue;
+
+      if (src.ptr.IsObject()) {
+        ObjDelete(src.ptr.Raw(), src.has_ttl);
+      } else {
+        auto& dest = items[dest_id++];
+        DenseLinkKey* link = src.ptr.AsLink();
+        dest.obj = link->Raw();
+        dest.has_ttl = link->HasTtl();
+        dest.ptr = link->next;
```

Comment:
Same here second time I see this maybe worth making it a function. Actually we have the exact same boilerplate in the body of the for-loop from line 277 to 297. See https://github.com/dragonflydb/dragonfly/pull/3862/files#diff-5685b15a1d75a2c80b43908203ee74ab673f9a748f1dce35ad1f018bb656eb6aR236 with the only difference the function that gets called `clone` vs `ObjDelete`

Reply:
it's not the same code because here we also call `FreeLink`